core/blockchain.go

```diff
@@ -672,8 +672,8 @@ func (bc *BlockChain) setHeadBeyondRoot(head uint64, time uint64, root common.Ha
 				beyondRoot := (root == common.Hash{}) // Flag whether we're beyond the requested root (no root, always true)
 
 				for {
-					// If a root threshold was requested but not yet crossed, check
-					if root != (common.Hash{}) && !beyondRoot && newHeadBlock.Root() == root {
+					// Check if the root threshold is crossed
+					if !beyondRoot && newHeadBlock.Root() == root {
```

Comment:
If the reasoning is `beyondRoot := (root == common.Hash{})` -> " no nneed to check if `root != (common.Hash{})`, then that is not a good reason, because we're in a loop here, and `beyondRoot` is mutated during the loop, so the initial condition will not necessarily hold.

Reply:
My apology for the empty commit message.
I will make a habit of writing a good PR description, thank you.

### What should've been included in the commit message
`beyondRoot` serves as a one-time flag that sets `rootNumber` just once if root is given.
The check is saying that if root was given and the header rewinds to the root, set the `rootNumber` and never come back. It is the only place in the loop that `beyondRoot` is mutated; if `beyondRoot` is ever set to true, that flag will never flip.

`beyondRoot` is initialized with `root == common.Hash{})`, and `root` is never mutated in the loop.
Since `beyondRoot` is not mutated elsewhere in the loop, so I figured the check was redundant.

---
Having said that, one could argue that the redundant check adds some clarity to the code. If that argument is sound to maintainers, especially on these core parts, I fully respect the decision.

Reply:
> one could argue that the redundant check adds some clarity to the code

Yep, I'm inclined to go that way. Thanks for the effort though!